llvm/lib/CodeGen/MachineOutliner.cpp

```diff
@@ -593,7 +593,11 @@ void MachineOutliner::findCandidates(
     unsigned NumDiscarded = 0;
     unsigned NumKept = 0;
 #endif
-    for (const unsigned &StartIdx : RS.StartIndices) {
+    // Sort the start indices so that we can efficiently check if candidates
+    // overlap with each other in MachineOutliner::findCandidates().
+    SmallVector<unsigned> SortedStartIndices(RS.StartIndices);
```

Comment:
Why do we need to make a copy here? Can we directly sort RS.StartIndices?

Reply:
It is because the iterator for the outer loop (`RS`) is of const type and thus we cannot be directly sort `RS.StartIndices`.

Reply:
I think it's a good suggestion. Can you simply drop `const` in the outer loop so that you can sort them in place?

Reply:
I dropped the `const` and sort the `RS.StartIndices` in place.